services/productService.ts

```diff
@@ -0,0 +1,104 @@
+// services/productService.ts
+import axiosInstance from '@/lib/axiosConfig';
+import { Product } from '@/app/types/Product';
+
+//using hardcoded jwt token for now
+const token = ""
+
+// Fetch all products
+export const getAllProducts = async () => {
+  try {
+    const response = await axiosInstance.get("/products");
+    return response.data;
+  } catch (err) {
+    throw err;
+  }
+};
+
+// Fetch product by ID
+export const getProductById = async (id: string): Promise<Product> => {
+  const response = await axiosInstance.get(`/products/${id}`);
+  return response.data;
+};
+
+export const addProduct = async (product: any) => {
+    const formData = new FormData();
+    formData.append(
+      "product",
+      JSON.stringify({
+        name: product.name,
+        description: product.description,
+        price: parseFloat(product.price),
+        stock: parseInt(product.stock),
+        category: product.category,
+      })
+    );
+  
+    if (product.image) {
+      formData.append("image", product.image);
+    }
+  
+    const response = await axiosInstance.post("/products", formData, {
+      headers: {
+        "Content-Type": "multipart/form-data",
+        Authorization: `Bearer ${token}`,
+      },
+    });
+  
+    return response.data;
+  };
+
+// Update a product
```

Comment:
```suggestion
```
@aaishahhamdha You don't have to use comments for things that are very explicit. I would suggest removing comments which are not necessary . Check other places too